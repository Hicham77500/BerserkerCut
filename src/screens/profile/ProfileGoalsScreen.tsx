import React, { useState } from 'react';
import { ScrollView, Text, StyleSheet, View } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { useAuth } from '../../hooks/useAuth';
import { Button, Card } from '../../components';
import { Colors, Typography, Spacing, BorderRadius } from '../../utils/theme';

const OBJECTIVE_OPTIONS = [
  { key: 'cutting', label: 'SÃ¨che' },
  { key: 'recomposition', label: 'Recomposition' },
  { key: 'maintenance', label: 'Maintien' },
] as const;

type ObjectiveKey = (typeof OBJECTIVE_OPTIONS)[number]['key'];

export const ProfileGoalsScreen: React.FC = () => {
  const { user, updateProfile } = useAuth();
  const currentObjective = (user?.profile?.objective ?? 'cutting') as ObjectiveKey;
  const [objective, setObjective] = useState<ObjectiveKey>(currentObjective);
  const [loading, setLoading] = useState(false);

  const handleSave = async () => {
    setLoading(true);
    try {
      await updateProfile({ objective });
    } finally {
      setLoading(false);
    }
  };

  return (
    <SafeAreaView style={styles.safeArea}>
      <ScrollView contentContainerStyle={styles.content}>
        <Card style={styles.card}>
          <Text style={styles.cardTitle}>Objectif principal</Text>
          <View style={styles.options}>
            {OBJECTIVE_OPTIONS.map((option) => {
              const active = option.key === objective;
              return (
                <Button
                  key={option.key}
                  title={option.label}
                  variant={active ? 'primary' : 'outline'}
                  onPress={() => setObjective(option.key)}
                />
              );
            })}
          </View>
          <Button title="Enregistrer" onPress={handleSave} loading={loading} />
        </Card>
      </ScrollView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  safeArea: {
    flex: 1,
    backgroundColor: Colors.background,
  },
  content: {
    flexGrow: 1,
    padding: Spacing.lg,
    gap: Spacing.lg,
  },
  card: {
    padding: Spacing.lg,
    borderRadius: BorderRadius.lg,
    backgroundColor: Colors.surface,
  },
  cardTitle: {
    ...Typography.h3,
    marginBottom: Spacing.md,
    color: Colors.text,
  },
  options: {
    gap: Spacing.sm,
    marginBottom: Spacing.lg,
  },
});

export default ProfileGoalsScreen;
