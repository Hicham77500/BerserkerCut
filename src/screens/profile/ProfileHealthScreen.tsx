import React, { useState, useMemo } from 'react';
import { ScrollView, View, Text, StyleSheet, Alert } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { useAuth } from '@/hooks/useAuth';
import { Input, Button, Card, IOSButton } from '@/components';
import { Typography, Spacing, BorderRadius, ThemePalette } from '@/utils/theme';
import { useThemeMode } from '@/hooks/useThemeMode';
import { HealthProfile } from '@/types';

export const ProfileHealthScreen: React.FC = () => {
  const { user, updateProfile } = useAuth();
  const { colors } = useThemeMode();
  const defaultHealth: HealthProfile = {
    weight: 0,
    height: 0,
    age: 0,
    gender: 'male',
    activityLevel: 'moderate',
    averageSleepHours: 8,
    dataSource: {
      type: 'manual',
      isConnected: false,
      permissions: [],
    },
    lastUpdated: new Date(),
    isManualEntry: true,
  };

  const health = user?.profile?.health ?? defaultHealth;

  const [form, setForm] = useState({
    weight: String(health.weight || ''),
    height: String(health.height || ''),
    age: String(health.age || ''),
    averageSleepHours: String(health.averageSleepHours || ''),
  });
  const [loading, setLoading] = useState(false);

  const handleSave = async () => {
    setLoading(true);
    try {
      await updateProfile({
        health: {
          ...health,
          weight: Number(form.weight) || 0,
          height: Number(form.height) || 0,
          age: Number(form.age) || 0,
          averageSleepHours: Number(form.averageSleepHours) || 0,
          lastUpdated: new Date(),
        },
      });
    } finally {
      setLoading(false);
    }
  };

  const styles = useMemo(() => createStyles(colors), [colors]);

  const handleSaveSuccess = () => {
    Alert.alert('Profil mis à jour', 'Vos données biométriques ont été enregistrées avec succès.');
  };

  return (
    <SafeAreaView edges={['left', 'right', 'bottom']} style={styles.safeArea}>
      <ScrollView contentContainerStyle={styles.content}>
        <Card style={styles.card}>
          <Text style={styles.cardTitle}>Mesures biométriques</Text>
          
          <Input
            label="Poids (kg)"
            value={form.weight}
            keyboardType="decimal-pad"
            testID="profile-health-weight"
            onChangeText={(weight) => setForm((prev) => ({ ...prev, weight }))}
          />
          <Text style={styles.helpText}>Votre poids actuel en kilogrammes</Text>
          
          <Input
            label="Taille (cm)"
            value={form.height}
            keyboardType="decimal-pad"
            testID="profile-health-height"
            onChangeText={(height) => setForm((prev) => ({ ...prev, height }))}
          />
          <Text style={styles.helpText}>Votre taille en centimètres</Text>
          
          <Input
            label="Âge"
            value={form.age}
            keyboardType="number-pad"
            testID="profile-health-age"
            onChangeText={(age) => setForm((prev) => ({ ...prev, age }))}
          />
          
          <Input
            label="Sommeil moyen (h)"
            value={form.averageSleepHours}
            keyboardType="decimal-pad"
            testID="profile-health-sleep"
            onChangeText={(value) => setForm((prev) => ({ ...prev, averageSleepHours: value }))}
          />
          <Text style={styles.helpText}>Nombre d'heures de sommeil par nuit en moyenne</Text>
          
          <View style={styles.activityContainer}>
            <Text style={styles.activityTitle}>Niveau d'activité</Text>
            <Text style={styles.helpText}>Sélectionnez le niveau qui correspond le mieux à votre mode de vie actuel</Text>
            {/* TODO: Implement activity level selection */}
          </View>

          <IOSButton label="Enregistrer" onPress={() => {
            handleSave().then(handleSaveSuccess).catch(error => {
              Alert.alert('Erreur', 'Une erreur est survenue lors de la sauvegarde. Veuillez réessayer.');
            });
          }} />
        </Card>
      </ScrollView>
    </SafeAreaView>
  );
};

const createStyles = (colors: ThemePalette) =>
  StyleSheet.create({
    safeArea: {
      flex: 1,
      backgroundColor: colors.background,
    },
    content: {
      flexGrow: 1,
      padding: Spacing.lg,
      gap: Spacing.md,
    },
    card: {
      borderRadius: BorderRadius.lg,
      padding: Spacing.lg,
      backgroundColor: colors.surface,
      gap: Spacing.md,
    },
    cardTitle: {
      ...Typography.h3,
      marginBottom: Spacing.md,
      color: colors.text,
    },
    row: {
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center',
      marginBottom: Spacing.sm,
    },
    radioContainer: {
      flexDirection: 'row',
      marginTop: Spacing.sm,
      marginBottom: Spacing.md,
      flexWrap: 'wrap',
      gap: Spacing.sm,
    },
    activityContainer: {
      marginBottom: Spacing.md,
    },
    activityTitle: {
      ...Typography.body,
      fontWeight: '600',
      color: colors.text,
      marginBottom: Spacing.sm,
    },
    helpText: {
      ...Typography.caption,
      color: colors.textLight,
      marginTop: -Spacing.xs,
      marginBottom: Spacing.sm,
    },
  });

export default ProfileHealthScreen;
