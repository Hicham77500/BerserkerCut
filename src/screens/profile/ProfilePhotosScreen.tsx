import React, { useEffect, useState, useCallback, useMemo, useRef } from 'react';
import {
  ScrollView,
  View,
  Text,
  StyleSheet,
  Image,
  Alert,
  TouchableOpacity,
  Switch,
  LayoutAnimation,
  Platform,
  UIManager,
} from 'react-native';
import * as ImagePicker from 'expo-image-picker';
import { Card, Button } from '@/components';
import { Typography, Spacing, BorderRadius, ThemePalette } from '@/utils/theme';
import { getSecureItem, setSecureItem } from '@/utils/storage/secureStorage';
import { CLOUD_CONSENT_STORAGE_KEY } from '@/constants/storageKeys';
import PhotoService from '@/services/photo';
import { useThemeMode } from '@/hooks/useThemeMode';
import { SafeAreaView } from 'react-native-safe-area-context';
import photoStorage, { StoredPhoto } from '@/services/photoStorage';
import { useFocusEffect } from '@react-navigation/native';

if (Platform.OS === 'android' && UIManager.setLayoutAnimationEnabledExperimental) {
  UIManager.setLayoutAnimationEnabledExperimental(true);
}

const MAX_PHOTOS = photoStorage.MAX_GALLERY_PHOTOS;

export const ProfilePhotosScreen: React.FC = () => {
  const { colors } = useThemeMode();
  const styles = useMemo(() => createStyles(colors), [colors]);
  const [photos, setPhotos] = useState<StoredPhoto[]>([]);
  const [isRequestingPermission, setIsRequestingPermission] = useState(false);
  const [cloudConsent, setCloudConsent] = useState(false);
  const [syncingCloud, setSyncingCloud] = useState(false);
  const [cloudStatus, setCloudStatus] = useState<string | null>(null);
  const photosRef = useRef<StoredPhoto[]>([]);

  useFocusEffect(
    useCallback(() => {
      let isActive = true;
      (async () => {
        const stored = await photoStorage.loadGallery();
        const consentValue = await getSecureItem(CLOUD_CONSENT_STORAGE_KEY);
        if (isActive) {
          setPhotos(stored);
          setCloudConsent(consentValue === 'true');
        }
      })();
      return () => {
        isActive = false;
      };
    }, [])
  );

  useEffect(() => {
    photosRef.current = photos;
  }, [photos]);

  const syncFromCloud = useCallback(
    async (localFallback?: StoredPhoto[], force = false) => {
      if (!cloudConsent && !force) return;
      setSyncingCloud(true);
      setCloudStatus('Synchronisation iCloud en cours…');
      try {
        const remote = await PhotoService.listPhotos();
        if (!remote || remote.length === 0) {
          setCloudStatus('Aucune photo iCloud disponible');
          return;
        }
        const baseSource = localFallback && localFallback.length ? localFallback : photosRef.current;
        const baseList = baseSource.length ? baseSource : await photoStorage.loadGallery();
        const mergedMap = new Map<string, StoredPhoto>();

        baseList.forEach((item) => {
          const key = item.id ?? `local-${item.timestamp}`;
          mergedMap.set(key, item);
        });

        for (const record of remote) {
          const identifier = record.id ?? `remote-${record.timestamp}`;
          const timestamp = record.timestamp ?? Date.now();
          const sourceUri = record.localUri ?? record.uri;
          const managedUri = await photoStorage.ensureLocalCopy(sourceUri, identifier);

          const existing = mergedMap.get(identifier);
          mergedMap.set(identifier, {
            ...(existing ?? {}),
            id: identifier,
            uri: managedUri,
            timestamp,
            cloudSynced: true,
          });
        }

        const trimmed = Array.from(mergedMap.values())
          .sort((a, b) => a.timestamp - b.timestamp)
          .slice(-MAX_PHOTOS);
        LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
        setPhotos(trimmed);
        await photoStorage.setGallery(trimmed);
        setCloudStatus('Synchronisation iCloud terminée');
      } catch (error) {
        console.warn('[ProfilePhotosScreen] syncFromCloud error', error);
        setCloudStatus("Erreur lors de la synchronisation avec iCloud");
      } finally {
        setSyncingCloud(false);
        setTimeout(() => setCloudStatus(null), 2000);
      }
    },
    [cloudConsent]
  );

  useEffect(() => {
    (async () => {
      const stored = await photoStorage.loadGallery();
      setPhotos(stored);
      const consentValue = await getSecureItem(CLOUD_CONSENT_STORAGE_KEY);
      const consentEnabled = consentValue === 'true';
      setCloudConsent(consentEnabled);
      if (consentEnabled) {
        await syncFromCloud(stored);
      }
    })();
  }, [syncFromCloud]);

  const requestCameraPermission = async (): Promise<boolean> => {
    setIsRequestingPermission(true);
    try {
      const { status } = await ImagePicker.requestCameraPermissionsAsync();
      if (status !== ImagePicker.PermissionStatus.GRANTED) {
        Alert.alert(
          'Permission refusée',
          'La caméra est nécessaire pour prendre des photos. Vous pouvez modifier cela dans les réglages.'
        );
        return false;
      }
      return true;
    } finally {
      setIsRequestingPermission(false);
    }
  };

  const handleTakePhoto = async () => {
    const hasPermission = await requestCameraPermission();
    if (!hasPermission) return;

    const result = await ImagePicker.launchCameraAsync({
      quality: 0.6,
      allowsEditing: true,
    });

    if (result.canceled || !result.assets?.length) {
      return;
    }

    const timestamp = Date.now();
    const identifier = `profile-${timestamp}`;
    const managedUri = await photoStorage.ensureLocalCopy(result.assets[0].uri, identifier);

    const newPhoto: StoredPhoto = {
      id: identifier,
      uri: managedUri,
      timestamp,
    };

    LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
    const trimmed = await photoStorage.pushToGallery(newPhoto);
    setPhotos(trimmed);

    if (cloudConsent) {
      setSyncingCloud(true);
      setCloudStatus('Synchronisation iCloud en cours…');
      PhotoService.uploadPhoto(newPhoto)
        .then((record) => {
          if (!record) return;
          setPhotos((prev) => {
            const updated = prev.map((item) =>
              item.timestamp === newPhoto.timestamp
                ? { ...item, id: record.id, cloudSynced: true }
                : item
            );
            void photoStorage.setGallery(updated);
            return updated;
          });
          setCloudStatus('Photo synchronisée sur iCloud');
        })
        .catch((error) => {
          console.warn('[ProfilePhotosScreen] upload error', error);
          setCloudStatus("Upload iCloud différé");
        })
        .finally(() => {
          setSyncingCloud(false);
          setTimeout(() => setCloudStatus(null), 2000);
        });
    }
  };

  const handleDeletePhoto = (timestamp: number) => {
    Alert.alert('Supprimer la photo ?', 'Cette action est irréversible.', [
      { text: 'Annuler', style: 'cancel' },
      {
        text: 'Supprimer',
        style: 'destructive',
        onPress: async () => {
          LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
          const { next, removed } = await photoStorage.removeFromGallery(timestamp);
          setPhotos(next);
          if (removed?.id && cloudConsent) {
            PhotoService.deletePhoto(removed.id).catch((error) =>
              console.warn('[ProfilePhotosScreen] delete error', error)
            );
          }
          await photoStorage.removeMealPhotosByTimestamp(timestamp);
        },
      },
    ]);
  };

  const handleClearAll = () => {
    Alert.alert('Effacer toutes les photos ?', 'Toutes vos photos locales seront supprimées.', [
      { text: 'Annuler', style: 'cancel' },
      {
        text: 'Supprimer',
        style: 'destructive',
        onPress: async () => {
          LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
          await photoStorage.clearAll();
          setPhotos([]);
        },
      },
    ]);
  };

  const handleConsentToggle = async (nextValue: boolean) => {
    if (nextValue) {
      Alert.alert(
        'Activer la synchronisation cloud ?',
        'Vos photos seront sauvegardées dans votre photothèque iCloud (album BerserkerCut). Vous pourrez retirer votre consentement à tout moment.',
        [
          { text: 'Annuler', style: 'cancel' },
          {
            text: 'Activer',
            onPress: async () => {
              await setSecureItem(CLOUD_CONSENT_STORAGE_KEY, 'true');
              setCloudConsent(true);
              setCloudStatus('Synchronisation en cours…');
              await syncFromCloud(photos, true);
            },
          },
        ]
      );
      return;
    }

    await setSecureItem(CLOUD_CONSENT_STORAGE_KEY, 'false');
    setCloudConsent(false);
    setCloudStatus(null);
  };

  return (
    <SafeAreaView style={styles.safeArea}>
      <ScrollView contentContainerStyle={styles.content}>
        <Card style={styles.headerCard}>
          <Text style={styles.title}>Suivi photo</Text>
          <Text style={styles.subtitle}>
            Vos photos sont stockées localement, chiffrées sur cet appareil. Elles ne sont jamais
            envoyées sur iCloud sans votre consentement explicite.
          </Text>
          <View style={styles.switchRow}>
            <View style={styles.switchLabels}>
              <Text style={styles.sectionTitle}>Synchronisation iCloud</Text>
              <Text style={styles.switchSubtitle}>
                Activez pour sauvegarder vos photos dans l'album iCloud BerserkerCut.
              </Text>
            </View>
            <Switch
              value={cloudConsent}
              onValueChange={handleConsentToggle}
              thumbColor={cloudConsent ? colors.primary : colors.textMuted}
              trackColor={{ true: colors.primary + '55', false: colors.border }}
            />
          </View>

          <Button
            title={isRequestingPermission ? 'Obtention de la permission...' : 'Prendre une photo'}
            onPress={handleTakePhoto}
            loading={isRequestingPermission}
            fullWidth
          />
          <Button
            title="Effacer toutes les photos locales"
            variant="outline"
            onPress={handleClearAll}
            fullWidth
            style={styles.clearButton}
          />
          {cloudConsent && (
            <Button
              title={syncingCloud ? 'Synchronisation…' : 'Rafraîchir depuis iCloud'}
              variant="ghost"
              onPress={() => syncFromCloud(photos)}
              fullWidth
              loading={syncingCloud}
            />
          )}
          {cloudStatus ? (
            <Text style={styles.statusText}>{cloudStatus}</Text>
          ) : null}
        </Card>

        <Card style={styles.listCard}>
          <Text style={styles.sectionTitle}>Historique (max {MAX_PHOTOS} photos)</Text>
          {photos.length === 0 ? (
            <Text style={styles.emptyText}>Aucune photo enregistrée pour le moment.</Text>
          ) : (
            photos
              .slice()
              .reverse()
              .map((photo) => (
                <View key={photo.timestamp} style={styles.photoRow}>
                  <Image source={{ uri: photo.uri }} style={styles.photo} />
                  <View style={styles.photoMeta}>
                    <Text style={styles.photoDate}>
                      {new Date(photo.timestamp).toLocaleString('fr-FR', {
                        day: '2-digit',
                        month: 'long',
                        year: 'numeric',
                        hour: '2-digit',
                        minute: '2-digit',
                      })}
                    </Text>
                    <TouchableOpacity onPress={() => handleDeletePhoto(photo.timestamp)}>
                      <Text style={styles.deleteLink}>Supprimer</Text>
                    </TouchableOpacity>
                  </View>
                </View>
              ))
          )}
        </Card>
      </ScrollView>
    </SafeAreaView>
  );
};

const createStyles = (colors: ThemePalette) =>
  StyleSheet.create({
    safeArea: {
      flex: 1,
      backgroundColor: colors.background,
    },
    content: {
      flexGrow: 1,
      padding: Spacing.lg,
      gap: Spacing.lg,
    },
    headerCard: {
      gap: Spacing.md,
      backgroundColor: colors.surface,
    },
    listCard: {
      gap: Spacing.md,
      backgroundColor: colors.surface,
    },
    switchRow: {
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'space-between',
      gap: Spacing.md,
    },
    switchLabels: {
      flex: 1,
      paddingRight: Spacing.md,
      gap: Spacing.xs,
    },
    title: {
      ...Typography.h2,
    },
    subtitle: {
      ...Typography.bodySmall,
      color: colors.textLight,
    },
    sectionTitle: {
      ...Typography.h3,
      color: colors.text,
    },
    switchSubtitle: {
      ...Typography.caption,
      color: colors.textLight,
    },
    emptyText: {
      ...Typography.body,
      color: colors.textLight,
    },
    photoRow: {
      flexDirection: 'row',
      alignItems: 'center',
      gap: Spacing.md,
    },
    photo: {
      width: 96,
      height: 96,
      borderRadius: BorderRadius.md,
      backgroundColor: colors.secondary,
    },
    photoMeta: {
      flex: 1,
      gap: Spacing.xs,
    },
    photoDate: {
      ...Typography.bodySmall,
      color: colors.text,
    },
    deleteLink: {
      ...Typography.caption,
      color: colors.error,
    },
    clearButton: {
      marginTop: Spacing.sm,
    },
    statusText: {
      ...Typography.caption,
      color: colors.textLight,
      textAlign: 'center',
    },
  });

export default ProfilePhotosScreen;
