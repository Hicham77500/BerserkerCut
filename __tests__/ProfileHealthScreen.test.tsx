import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import { Alert } from 'react-native';
import { ProfileHealthScreen } from '../src/screens/profile/ProfileHealthScreen';

const mockUpdateProfile = jest.fn().mockResolvedValue(undefined);

jest.mock('../src/hooks/useAuth', () => ({
  useAuth: () => ({
    user: {
      id: 'user-1',
      profile: {
        health: {
          weight: 75,
          height: 180,
          age: 30,
          gender: 'male',
          activityLevel: 'moderate',
          averageSleepHours: 7,
          dataSource: {
            type: 'manual',
            isConnected: false,
            permissions: [],
          },
          lastUpdated: new Date('2024-01-01T00:00:00Z'),
          isManualEntry: true,
        },
      },
    },
    updateProfile: mockUpdateProfile,
  }),
}));

jest.mock('../src/hooks/useThemeMode', () => {
  const { DarkColors, DarkNavigationTheme } = require('../src/utils/theme');
  return {
    useThemeMode: () => ({
      mode: 'dark',
      preference: 'dark',
      isDark: true,
      colors: DarkColors,
      toggleTheme: jest.fn(),
      setTheme: jest.fn(),
      navigationTheme: DarkNavigationTheme,
      ready: true,
      isSystemPreferred: false,
    }),
  };
});

describe('ProfileHealthScreen', () => {
  beforeEach(() => {
    mockUpdateProfile.mockClear();
    jest.spyOn(Alert, 'alert').mockImplementation(() => {});
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('soumet les valeurs converties lors de la sauvegarde et correspond au snapshot', async () => {
    const { getByTestId, getByText, toJSON } = render(<ProfileHealthScreen />);

    fireEvent.changeText(getByTestId('profile-health-weight'), '80');
    fireEvent.changeText(getByTestId('profile-health-height'), '182');
    fireEvent.changeText(getByTestId('profile-health-age'), '31');
    fireEvent.changeText(getByTestId('profile-health-sleep'), '7.5');

    fireEvent.press(getByText('Enregistrer'));

    await waitFor(() => {
      expect(mockUpdateProfile).toHaveBeenCalledWith(
        expect.objectContaining({
          health: expect.objectContaining({
            weight: 80,
            height: 182,
            age: 31,
            averageSleepHours: 7.5,
            lastUpdated: expect.any(Date),
          }),
        })
      );
    });

    expect(toJSON()).toMatchSnapshot();
  });
});
