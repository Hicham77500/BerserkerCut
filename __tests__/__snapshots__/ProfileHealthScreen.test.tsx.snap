// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileHealthScreen soumet les valeurs converties lors de la sauvegarde et correspond au snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "off",
    }
  }
  style={
    {
      "backgroundColor": "#1C1A18",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "flexGrow": 1,
        "gap": 16,
        "padding": 24,
      }
    }
  >
    <View>
      <View
        style={
          [
            {
              "backgroundColor": "#251C17",
              "borderRadius": 12,
              "overflow": "hidden",
            },
            {
              "elevation": 3,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.15,
              "shadowRadius": 3,
            },
            {
              "padding": 16,
            },
            {
              "backgroundColor": "#251C17",
              "borderRadius": 12,
              "gap": 16,
              "padding": 24,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#F5F2F0",
              "fontSize": 20,
              "fontWeight": "600",
              "letterSpacing": 0.15,
              "lineHeight": 28,
              "marginBottom": 16,
            }
          }
        >
          Mesures biométriques
        </Text>
        <View
          style={
            {
              "marginBottom": 16,
            }
          }
        >
          <Text
            style={
              {
                "color": "#F5F2F0",
                "fontSize": 15,
                "fontWeight": "500",
                "letterSpacing": -0.24,
                "lineHeight": 20,
                "marginBottom": 4,
              }
            }
          >
            Poids (kg)
          </Text>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "#251C17",
                  "borderRadius": 8,
                  "flexDirection": "row",
                },
                {
                  "borderColor": "#3E2F26",
                  "borderWidth": 1,
                  "elevation": 1,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 1,
                    "width": 0,
                  },
                  "shadowOpacity": 0.12,
                  "shadowRadius": 2,
                },
                {
                  "minHeight": 44,
                  "paddingHorizontal": 16,
                },
                false,
                undefined,
                undefined,
              ]
            }
          >
            <TextInput
              blurOnSubmit={true}
              keyboardType="decimal-pad"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholderTextColor="#A99280"
              returnKeyType="done"
              secureTextEntry={false}
              style={
                [
                  {
                    "color": "#F5F2F0",
                    "flex": 1,
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0.5,
                    "lineHeight": 24,
                  },
                ]
              }
              testID="profile-health-weight"
              value="80"
            />
          </View>
        </View>
        <Text
          style={
            {
              "color": "#C4BDB9",
              "fontSize": 12,
              "fontWeight": "400",
              "letterSpacing": 0.4,
              "lineHeight": 16,
              "marginBottom": 8,
              "marginTop": -4,
            }
          }
        >
          Votre poids actuel en kilogrammes
        </Text>
        <View
          style={
            {
              "marginBottom": 16,
            }
          }
        >
          <Text
            style={
              {
                "color": "#F5F2F0",
                "fontSize": 15,
                "fontWeight": "500",
                "letterSpacing": -0.24,
                "lineHeight": 20,
                "marginBottom": 4,
              }
            }
          >
            Taille (cm)
          </Text>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "#251C17",
                  "borderRadius": 8,
                  "flexDirection": "row",
                },
                {
                  "borderColor": "#3E2F26",
                  "borderWidth": 1,
                  "elevation": 1,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 1,
                    "width": 0,
                  },
                  "shadowOpacity": 0.12,
                  "shadowRadius": 2,
                },
                {
                  "minHeight": 44,
                  "paddingHorizontal": 16,
                },
                false,
                undefined,
                undefined,
              ]
            }
          >
            <TextInput
              blurOnSubmit={true}
              keyboardType="decimal-pad"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholderTextColor="#A99280"
              returnKeyType="done"
              secureTextEntry={false}
              style={
                [
                  {
                    "color": "#F5F2F0",
                    "flex": 1,
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0.5,
                    "lineHeight": 24,
                  },
                ]
              }
              testID="profile-health-height"
              value="182"
            />
          </View>
        </View>
        <Text
          style={
            {
              "color": "#C4BDB9",
              "fontSize": 12,
              "fontWeight": "400",
              "letterSpacing": 0.4,
              "lineHeight": 16,
              "marginBottom": 8,
              "marginTop": -4,
            }
          }
        >
          Votre taille en centimètres
        </Text>
        <View
          style={
            {
              "marginBottom": 16,
            }
          }
        >
          <Text
            style={
              {
                "color": "#F5F2F0",
                "fontSize": 15,
                "fontWeight": "500",
                "letterSpacing": -0.24,
                "lineHeight": 20,
                "marginBottom": 4,
              }
            }
          >
            Âge
          </Text>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "#251C17",
                  "borderRadius": 8,
                  "flexDirection": "row",
                },
                {
                  "borderColor": "#3E2F26",
                  "borderWidth": 1,
                  "elevation": 1,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 1,
                    "width": 0,
                  },
                  "shadowOpacity": 0.12,
                  "shadowRadius": 2,
                },
                {
                  "minHeight": 44,
                  "paddingHorizontal": 16,
                },
                false,
                undefined,
                undefined,
              ]
            }
          >
            <TextInput
              blurOnSubmit={true}
              keyboardType="number-pad"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholderTextColor="#A99280"
              returnKeyType="done"
              secureTextEntry={false}
              style={
                [
                  {
                    "color": "#F5F2F0",
                    "flex": 1,
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0.5,
                    "lineHeight": 24,
                  },
                ]
              }
              testID="profile-health-age"
              value="31"
            />
          </View>
        </View>
        <View
          style={
            {
              "marginBottom": 16,
            }
          }
        >
          <Text
            style={
              {
                "color": "#F5F2F0",
                "fontSize": 15,
                "fontWeight": "500",
                "letterSpacing": -0.24,
                "lineHeight": 20,
                "marginBottom": 4,
              }
            }
          >
            Sommeil moyen (h)
          </Text>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "#251C17",
                  "borderRadius": 8,
                  "flexDirection": "row",
                },
                {
                  "borderColor": "#3E2F26",
                  "borderWidth": 1,
                  "elevation": 1,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 1,
                    "width": 0,
                  },
                  "shadowOpacity": 0.12,
                  "shadowRadius": 2,
                },
                {
                  "minHeight": 44,
                  "paddingHorizontal": 16,
                },
                false,
                undefined,
                undefined,
              ]
            }
          >
            <TextInput
              blurOnSubmit={true}
              keyboardType="decimal-pad"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholderTextColor="#A99280"
              returnKeyType="done"
              secureTextEntry={false}
              style={
                [
                  {
                    "color": "#F5F2F0",
                    "flex": 1,
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0.5,
                    "lineHeight": 24,
                  },
                ]
              }
              testID="profile-health-sleep"
              value="7.5"
            />
          </View>
        </View>
        <Text
          style={
            {
              "color": "#C4BDB9",
              "fontSize": 12,
              "fontWeight": "400",
              "letterSpacing": 0.4,
              "lineHeight": 16,
              "marginBottom": 8,
              "marginTop": -4,
            }
          }
        >
          Nombre d'heures de sommeil par nuit en moyenne
        </Text>
        <View
          style={
            {
              "marginBottom": 16,
            }
          }
        >
          <Text
            style={
              {
                "color": "#F5F2F0",
                "fontSize": 16,
                "fontWeight": "600",
                "letterSpacing": 0.5,
                "lineHeight": 24,
                "marginBottom": 8,
              }
            }
          >
            Niveau d'activité
          </Text>
          <Text
            style={
              {
                "color": "#C4BDB9",
                "fontSize": 12,
                "fontWeight": "400",
                "letterSpacing": 0.4,
                "lineHeight": 16,
                "marginBottom": 8,
                "marginTop": -4,
              }
            }
          >
            Sélectionnez le niveau qui correspond le mieux à votre mode de vie actuel
          </Text>
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "borderRadius": 16,
                "elevation": 3,
                "justifyContent": "center",
                "minHeight": 48,
                "paddingHorizontal": 24,
                "paddingVertical": 16,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.15,
                "shadowRadius": 3,
              },
              {
                "backgroundColor": "#C2633A",
              },
              false,
              false,
              false,
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                {
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#F5F2F0",
                    "fontSize": 16,
                    "fontWeight": "600",
                    "letterSpacing": 0.5,
                    "lineHeight": 24,
                  },
                  {
                    "textAlign": "center",
                  },
                  {
                    "color": "#F5F2F0",
                  },
                ]
              }
            >
              Enregistrer
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
